cothink2/client/lib/helpers.js

```diff
@@ -0,0 +1,11 @@
+assert = function (cond, message) {
+    if (!cond) {
+        message = message || "Assertion failed.";
+        throw new Error(message);
+    }
+}
+
+assert_type = function (object, prototype) {
+    message = "Object " + object + " is not of type " + prototype + ".";
+    assert(prototype.isPrototypeOf(object), message);
+}
```

Comment:
Thanks for the tip, but simply `check(item, Item)` would not work: we would need to define something like `ItemFields = {id: String, _id: String}` to use for checking. It would allow duck typing but it would not check for the presence of methods, so it is probably not the right tool here.

I'd be happy to switch to other approaches, I think some type checking can be helpful to detect mistakes (something like Java's interfaces may be nicer than a class-like system though, which my `assert_type` solution tends towards).